drivers/gpio/Kconfig.adi_max22017

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024 Analog Devices Inc.
+# Copyright (c) 2024 BayLibre SAS
+# SPDX-License-Identifier: Apache-2.0
+
+config GPIO_MAX22017
+	bool "Analog Devices MAX22017 GPIO support"
+	default y
+	depends on DT_HAS_ADI_MAX22017_GPIO_ENABLED
+	select MFD
+	help
+	  Enable GPIO support for the Analog Devices MAX22017
+
+if GPIO_MAX22017
+
+config GPIO_MAX22017_INIT_PRIORITY
+	int "Init priority"
+	default 81
+	help
+	  Analog Devices MAX22017 gpio device driver initialization priority.
+
+config GPIO_MAX22017_INT_QUIRK
+	bool "MAX22017 GPIO Interrupt quirk"
+	default n
```

Comment:
As far as I understand, the driver doesn't work properly without the quirk, correct? If that's the case, should it not be `default y`? Otherwise, if you prefer to keep `default n`, this line should be deleted, as `n` is always the default.

Reply:
I think it mostly comes from my testing setup as I don't have a clean debounced input to try it... so any attempt at testing this out generates a storm of interrupts.
it "works" but sometimes can get stuck with a pending interrupt to read.
"polling" the interrupt register is the only way I have found to get out of that "stuck" situation.

maybe someone with a clean input signal will never reproduce the issues I'm seeing? that's what I'm hoping for a the rationale for the quirk beeing disabled by default.